exec/java-exec/src/main/java/org/apache/drill/exec/store/schedule/AssignmentCreator.java

```diff
@@ -50,7 +50,8 @@
   private static Comparator<Entry<DrillbitEndpoint,Long>> comparator = new Comparator<Entry<DrillbitEndpoint,Long>>() {
     @Override
     public int compare(Entry<DrillbitEndpoint, Long> o1, Entry<DrillbitEndpoint,Long> o2) {
-      return (int) (o1.getValue() - o2.getValue());
+      long ret = o1.getValue() - o2.getValue();
+      return ret > 0? 1 : ((ret < 0)? -1: 0);
```

Comment:
this doesn't work for `o1.getValue() == Long.MAX_VALUE` and `o2.getValue() == Long.MIN_VALUE`